admin/tracking/class-tracking.php

```diff
@@ -192,4 +191,33 @@ public function get_collector() {
 
 		return $collector;
 	}
+
+	/**
+	 * See if we should run tracking at all.
+	 */
+	private function tracking_enabled() {
+		// Check if we're allowing tracking.
+		$tracking = WPSEO_Options::get( 'tracking' );
+
+		if ( $tracking === false ) {
+			return;
+		}
+
+		/**
+		 * Filter: 'wpseo_enable_tracking' - Enables the data tracking of Yoast SEO Premium and add-ons.
+		 *
+		 * @api string $is_enabled The enabled state. Default is false.
+		 */
+		$filter = apply_filters( 'wpseo_enable_tracking', false );
+
+		// Save this state.
+		if ( $tracking === null ) {
+			$tracking = $filter;
```

Comment:
I'd run the filter here as it's value is only used if `$tracking` is `null.` Saves a bit of processing.

I would also expect `$tracking` to be assigned the value of `$filter` here.